### PR TITLE
Unenroll students on removal from last classroom

### PR DESCRIPTION
**Pull request by phoenixeliot**
When removing a student from a classroom, if the student is in no other classrooms:
- Inform the teacher that the license will be restored IF they have rights to the license
- Revoke the license

If the student is still in other classrooms, the old behavior still takes place — the student stays enrolled, and the teacher is told they will not get the license back.

### CODE REVIEW

File: server/middleware/classrooms.coffee [hunk 1]
```diff
@@ -143,6 +144,79 @@ module.exports =
...
+    unless ownsClassroom or removingSelf
+      throw new errors.Forbidden()
+
+    try
+      otherClassrooms = yield Classroom.find { members: mongoose.Types.ObjectId(userID), _id: {$ne: classroom.get('_id')} }
+    catch err
+      throw new errors.InternalServerError('Error finding other classrooms by memberID: ' + err)
+  
+    # If the student is being removed from their very last classroom, unenroll them
+    user = yield User.findOne({ _id: mongoose.Types.ObjectId(userID) })
```
Comment (differentmatt) [inline server/middleware/classrooms.coffee]: `findById` instead?

File: server/middleware/classrooms.coffee [hunk 2]
```diff
@@ -143,6 +144,79 @@ module.exports =
...
+    throw new errors.UnprocessableEntity('Member ID must be a MongoDB ID') unless utils.isID(userID)
+    try
+      classroom = yield Classroom.findById req.params.classroomID
+    catch err
+      throw new errors.InternalServerError('Error finding classroom by ID: ' + err)
+    throw new errors.NotFound('No classroom found with that ID') if not classroom
+    if not _.any(classroom.get('members'), (memberID) -> memberID.toString() is userID)
+      throw new errors.Forbidden()
+    ownsClassroom = classroom.get('ownerID').equals(req.user.get('_id'))
+    removingSelf = userID is req.user.id
```
Comment (differentmatt) [inline server/middleware/classrooms.coffee]: Students can't remove themselves from classroom I don't think.

File: server/models/Prepaid.coffee [hunk 3]
```diff
@@ -47,6 +47,13 @@ PrepaidSchema.methods.canReplaceUserPrepaid = (otherPrepaid) ->
 PrepaidSchema.methods.canBeUsedBy = (userID) ->
   @get('creator').equals(userID) or _.find(@get('joiners'), (joiner) -> joiner.userID.equals(userID) )
 
+PrepaidSchema.methods.wasStudentEnrolledByTeacher = (userID, teacherID) ->
```
Comment (differentmatt) [inline server/models/Prepaid.coffee]: Does this need a course type check similar to revoke?

File: server/middleware/classrooms.coffee (hunk 1 above)
Comment (phoenixeliot) [reply server/middleware/classrooms.coffee]: Ah, yeah, I forget about that. I'll leave it so I don't have to test again but I'll try to remember that method exists.

File: server/models/Prepaid.coffee (hunk 3 above)
Comment (phoenixeliot) [reply server/models/Prepaid.coffee]: Ah, yup, good catch.

**Pull request merged**